feat(export-transactions): implement in-app purchase

JosephSanjaya opened the pull request:
## 📱 Description
- Integrate with revenue cat
- Implement csv exporter and launcher
- Implement pdf renderer and exporter

## Platform
- [x] Android
- [ ] iOS
- [ ] Games-Unity
- [ ] DevOps (AWS)
- [ ] Website
- [ ] C/Golang 

## 🎯 Type of Change
<!-- Mark the relevant option with an [x] -->
- [ ] 🐛 Bug fix
- [x] ✨ New feature
- [ ] 🔧 Refactoring (code change that neither fixes a bug nor adds a feature)
- [ ] 📚 Documentation update
- [ ] 🎨 UI/UX improvement
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test addition or improvement


## 🎯 Steps to reproduce and/or Test
1. Please always update submodules first
2. If you pull from [main repository](https://github.com/gruntsoftware/android), make sure the general purpose and the main repo on branch `js/feat/iap`
3. Sync the projects, you should see 2 app configuration `app` and `general-purpose-app`
4. both can be run `app` will launch normal apps, but general-purpose-app will launch test app for faster debugging.
5. Use account that has transactions, if not then the button to export should not shown
6. Click export then continue the flow


## 📋 Changes
- Integrate with revenue cat
- Implement csv exporter and launcher
- Implement pdf renderer and exporter

### 🔗 Related Issues
<!-- Link any related issues using "Fixes #issue_number" or "Closes #issue_number" -->
- Related to https://github.com/gruntsoftware/internal/issues/239
- Related to https://github.com/gruntsoftware/internal/issues/238
- Related to https://github.com/gruntsoftware/internal/issues/191

## 🧪 Tests Status
- [ ] Tests ran successfully locally?
- [ ] Added more tests? How many?
- [ ] Code coverage percentage of the codebase: __%


Uploading CleanShot 2025-10-06 at 00.34.13.mp4…


## 🎯 Reviewers
<!-- Tag specific reviewers or teams -->
@kcw-grunt, @josikie

kcw-grunt committed:
create private gp submodule

kcw-grunt committed:
Update settings.gradle.kts

JosephSanjaya committed:
chore: Migrate build logic to version catalog and update submodules

This commit migrates the project's build logic to use a centralized version catalog (`grunt`) and updates submodule references.

Specific changes include:

*   **Submodule Updates:**
    *   Updated `gruntsoftware-build-logic` submodule pointer.
    *   Updated `modules/private-general-purpose` submodule pointer.
*   **Build Configuration (`settings.gradle.kts`):**
    *   Removed the conditional loading of the `games` version catalog.
    *   Added a `grunt` version catalog loaded from `gruntsoftware-build-logic/gradle/libs.versions.toml`.
    *   Changed the included module from `:modules:private-general-purpose:content` to `:modules:private-general-purpose:games`.
    *   Removed conditional inclusion logic for `:modules:private-general-purpose:content`.
*   **Root `build.gradle.kts`:**
    *   Updated plugin aliases from `libs.plugins.android.*` to `grunt.plugins.android.*`.
*   **App `build.gradle.kts`:**
    *   Updated plugin alias for `com.android.application` from `libs` to `grunt`.
    *   Updated the `games` module dependency to `implementation(project(":modules:private-general-purpose:games"))` and removed the conditional logic for `findProject(":modules:games:content")`.
    *   Updated various dependencies to use the `grunt` version catalog (e.g., `grunt.androidx.core.ktx`, `grunt.androidx.compose.bom`, `grunt.bundles.androidx.compose`, `grunt.material`).
*   **Version Catalog (`gradle/libs.versions.toml`):**
    *   Removed versions for `agp` and `androidx-core-ktx`.
    *   Removed versions for `androidx-compose-bom`, `androidx-activity-compose`, and `androidx-fragment-compose`.
    *   Removed `androidx-core` library alias.
    *   Removed various `androidx-compose` library aliases and the `androidx-compose` bundle.
    *   Removed `android-application` and `android-library` plugin aliases.

JosephSanjaya committed:
feat(deps): add kotlin-immutable dependency

This commit adds the `kotlinx-collections-immutable` library, version 0.4.0, to the project's dependencies.

josikie commented:
Amazing, I'll test at it, also please resolve the conflicts, thank you! @JosephSanjaya

kcw-grunt commented:
Thanks @JosephSanjaya . Once the general purpose repo is tested and approved I will focus on this PR.

kcw-grunt reviewed (changes requested):
Just need rebase (the private and build repo) before reviewing @JosephSanjaya

JosephSanjaya commented:
Rebased!

kcw-grunt commented on app/src/main/java/com/brainwallet/ui/screens/home/composable/HomeSettingDrawerSheet.kt:
```diff
@@ -42,7 +42,7 @@ import com.brainwallet.ui.screens.home.composable.settingsrows.SecurityDetail
 import com.brainwallet.ui.screens.home.composable.settingsrows.SettingRowItem
 import com.brainwallet.ui.screens.home.composable.settingsrows.ThemeSettingRowItem
 import com.brainwallet.ui.theme.BrainwalletAppTheme
```
Why are we switching the namespace from 
`com.brainwallet` vs `ltd.grunt.brainwallet`  @JosephSanjaya ?

JosephSanjaya commented:
Add simple test, and fix failed pipeline @kcw-grunt, PR is ready

kcw-grunt commented:
> Add simple test, and fix failed pipeline @kcw-grunt, PR is ready
Yup Im running through sync and func test @JosephSanjaya Did you see my question re namespace?

JosephSanjaya commented:
> > Add simple test, and fix failed pipeline @kcw-grunt, PR is ready
> > Yup Im running through sync and func test @JosephSanjaya Did you see my question re namespace?

did you mean the chilli one?

JosephSanjaya commented:
ohh that color is moved to core modules on general purpose, hmm because that already merged i will fix that  on next PR on general purpose repo is that okay? @kcw-grunt ?

kcw-grunt commented:
> > > Add simple test, and fix failed pipeline @kcw-grunt, PR is ready
> > > Yup Im running through sync and func test @JosephSanjaya Did you see my question re namespace?
> 
> did you mean the chilli one?

No I meant this :
<img width="1642" height="666" alt="Screenshot 2025-10-07 at 15 44 23" src="https://github.com/user-attachments/assets/d330b5cc-ba13-46f3-a853-93d3acd13c08" />

JosephSanjaya commented:
> > > > Add simple test, and fix failed pipeline @kcw-grunt, PR is ready
> > > > Yup Im running through sync and func test @JosephSanjaya Did you see my question re namespace?
> > 
> > 
> > did you mean the chilli one?
> 
> No I meant this : <img alt="Screenshot 2025-10-07 at 15 44 23" width="1642" height="666" src="https://private-user-images.githubusercontent.com/2899463/498364382-d330b5cc-ba13-46f3-a853-93d3acd13c08.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk4NDg5MTAsIm5iZiI6MTc1OTg0ODYxMCwicGF0aCI6Ii8yODk5NDYzLzQ5ODM2NDM4Mi1kMzMwYjVjYy1iYTEzLTQ2ZjMtYTg1My05M2QzYWNkMTNjMDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDdUMTQ1MDEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzkyNjMyNmU0ZjUwN2IwNjk5OTEyZjI0ZDhlYjc4ZWE3MDhlYzcyMTZkNmNiODZjZjA4MThlY2EzNjA2ZDE3OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.0bntA8EKKEkbv-pBg84c6GstOgMsDRosG0isob6NyTk">

that was also coming from general purpose repo. i move the theme for reusability . i think that was the namespace at the beginning of games repo.

i will change it on next PR if you want me to change that. @kcw-grunt

kcw-grunt commented:
> > > > > Add simple test, and fix failed pipeline @kcw-grunt, PR is ready
> > > > > Yup Im running through sync and func test @JosephSanjaya Did you see my question re namespace?
> > > 
> > > 
> > > did you mean the chilli one?
> > 
> > 
> > No I meant this : <img alt="Screenshot 2025-10-07 at 15 44 23" width="1642" height="666" src="https://private-user-images.githubusercontent.com/2899463/498364382-d330b5cc-ba13-46f3-a853-93d3acd13c08.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk4NDg5MTAsIm5iZiI6MTc1OTg0ODYxMCwicGF0aCI6Ii8yODk5NDYzLzQ5ODM2NDM4Mi1kMzMwYjVjYy1iYTEzLTQ2ZjMtYTg1My05M2QzYWNkMTNjMDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDdUMTQ1MDEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzkyNjMyNmU0ZjUwN2IwNjk5OTEyZjI0ZDhlYjc4ZWE3MDhlYzcyMTZkNmNiODZjZjA4MThlY2EzNjA2ZDE3OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.0bntA8EKKEkbv-pBg84c6GstOgMsDRosG0isob6NyTk">
> 
> that was also coming from general purpose repo. i move the theme for reusability . i think that was the namespace at the beginning of games repo.
> 
> i will change it on next PR if you want me to change that. @kcw-grunt

Thats fine. I'm less worried about that...I just wanted to understand. 
But having `chili` needs to be fixed because all of our assets are already using that spelling. Looking forward to seeing that taken care of next time @JosephSanjaya

JosephSanjaya commented:
> > > > > > Add simple test, and fix failed pipeline @kcw-grunt, PR is ready
> > > > > > Yup Im running through sync and func test @JosephSanjaya Did you see my question re namespace?
> > > > 
> > > > 
> > > > did you mean the chilli one?
> > > 
> > > 
> > > No I meant this : <img alt="Screenshot 2025-10-07 at 15 44 23" width="1642" height="666" src="https://private-user-images.githubusercontent.com/2899463/498364382-d330b5cc-ba13-46f3-a853-93d3acd13c08.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk4NDg5MTAsIm5iZiI6MTc1OTg0ODYxMCwicGF0aCI6Ii8yODk5NDYzLzQ5ODM2NDM4Mi1kMzMwYjVjYy1iYTEzLTQ2ZjMtYTg1My05M2QzYWNkMTNjMDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDdUMTQ1MDEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzkyNjMyNmU0ZjUwN2IwNjk5OTEyZjI0ZDhlYjc4ZWE3MDhlYzcyMTZkNmNiODZjZjA4MThlY2EzNjA2ZDE3OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.0bntA8EKKEkbv-pBg84c6GstOgMsDRosG0isob6NyTk">
> > 
> > 
> > that was also coming from general purpose repo. i move the theme for reusability . i think that was the namespace at the beginning of games repo.
> > i will change it on next PR if you want me to change that. @kcw-grunt
> 
> Thats fine. I'm less worried about that...I just wanted to understand. But having `chili` needs to be fixed because all of our assets are already using that spelling. Looking forward to seeing that taken care of next time @JosephSanjaya

Okay Noted @kcw-grunt , i think that name is carried out from main repo, because im not changing anything, just moving the places. will fix it later

JosephSanjaya committed:
refactor(app): Relocate core module imports

Updates the import paths for the core presentation module from `ltd.grunt.brainwallet.core.presentation` to `com.grunt.brainwallet.core.presentation`. This refactoring touches multiple UI components and initializers across the `app` module to align with the new package structure.

Additionally, this commit corrects a typo in `SeedWordItem.kt`, changing the error color variable from `chilli` to `chili` to match the theme definition.

Finally, the submodule pointer for `private-general-purpose` is updated.

JosephSanjaya commented:
Change branch on general-purpose, it should be ready to check @kcw-grunt

kcw-grunt reviewed (approved):
LGTM

josikie reviewed (approved):
Awesome, thanks @JosephSanjaya
